src/Orchard.Web/Modules/Orchard.OutputCache/Filters/OutputCacheFilter.cs

```diff
@@ -386,7 +386,7 @@ public class OutputCacheFilter : FilterProvider, IActionFilter, IResultFilter, I
 
             // Vary by authentication state if configured.
             if (CacheSettings.VaryByAuthenticationState) {
-                result["auth"] = filterContext.HttpContext.User.Identity.IsAuthenticated.ToString().ToLowerInvariant();
+                result["auth"] = result["auth"] = _workContext.CurrentUser != null;
```

Comment:
there is only one result["auth"] assignment. The line in pink is replaced by the line in green.